lib/src/main/java/com/segment/analytics/kotlin/destinations/amplitude/AmplitudeSession.kt

```diff
@@ -80,20 +83,23 @@ class AmplitudeSession (private val sessionTimeoutMs : Long = 300000) : EventPlu
     }
 
     private fun startNewSessionIfNecessary() {
-        val current = java.lang.System.currentTimeMillis()
-        // Make sure the first event has a valid ID and we send a session start.
-        // Subsequent events should have session IDs updated after the session track event is sent
-        if(eventSessionId == -1L || sessionId == -1L) {
-            sessionId = current
-            eventSessionId = current
-            startNewSession()
-        } else if (current - lastEventFiredTime >= sessionTimeoutMs) {
-            sessionId = current
-
-            endSession()
-            startNewSession()
+        sessionUpdateLock.withLock {
```

Comment:
since this forces synchronization on a single method, we can probably do it in the old java way? add a `synchronized` modifier to the method (see [this](https://discuss.kotlinlang.org/t/synchronized-to-modifier/87)). not sure if it's better though

Reply:
Kotlin considers it deprecated.